helm/templates/ingress.yaml

```diff
@@ -0,0 +1,32 @@
+{{- if .Values.coder.ingress.enable }}
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
```

Comment:
Do we need to provide an ingress? Would a service be enough? I think @f0ssel had some thoughts here.

Reply:
The Ingress is useful for customers who don't want to expose coder directly with a LoadBalancer Service but instead want to run it through their own ingress controller like the nginx one. Turning on the ingress does not create an ingress controller for you like it did in v1 at one point, the customer needs to supply their own.

Reply:
Would it be simpler to exclude that resource for now, as customers can *really easily* add their own Ingress?

My intent is to minimize the surface area and knowledge of Kubernetes that we need to have. Reducing resources that some devs may not understand can be helpful there.

Reply:
Sounds good to me, I'm down to remove it if you're fine with that in the name of simplicity

Reply:
I'm good with that if you feel it's not necessary, I don't have a full understanding, I'm just tryna simplify ;p

Reply:
@f0ssel you chill with me removing ingress?

Reply:
ultra chill 👍

Reply:
It will be required for us to document the step of optionally creating an ingress on setup instructions since it will be a super common ask, but I'm fine with it not being a part of coder chart. This will also mean "rolling out changes" to an ingress will also be more complicated to instruct users, but I don't see that being something we really need to worry about imo.